Source/Scene/PointCloud3DTileContent.js

```diff
@@ -823,58 +713,310 @@ define([
+        var isQuantized = content._isQuantized;
+        var isOctEncoded16P = content._isOctEncoded16P;
+        var isRGB565 = content._isRGB565;
+        var isTranslucent = content._isTranslucent;
+        var hasColors = content._hasColors;
+        var hasNormals = content._hasNormals;
+        var hasBatchIds = content._hasBatchIds;
+        var backFaceCulling = content._backFaceCulling;
+        var vertexArray = content._drawCommand.vertexArray;
+
+        var colorStyleFunction;
+        var showStyleFunction;
+        var styleTranslucent = isTranslucent;
+
+        if (hasBatchTable) {
+            // Styling is handled in the batch table
+            hasStyle = false;
+        }
+
+        if (hasStyle) {
+            var shaderState = {
+                translucent : false
+            };
+            colorStyleFunction = style.getColorShaderFunction('getColorFromStyle', 'czm_tiles3d_style_', shaderState);
+            showStyleFunction = style.getShowShaderFunction('getShowFromStyle', 'czm_tiles3d_style_', shaderState);
+            styleTranslucent = shaderState.translucent;
+        }
+
+        content._styleTranslucent = styleTranslucent;
+
+        var hasColorStyle = defined(colorStyleFunction);
+        var hasShowStyle = defined(showStyleFunction);
+
+        // Get the properties in use by the style
+        var styleableProperties = [];
+        if (hasColorStyle) {
+            getStyleableProperties(colorStyleFunction, styleableProperties);
+        }
+        if (hasShowStyle) {
+            getStyleableProperties(showStyleFunction, styleableProperties);
+        }
+
+        // Disable vertex attributes that aren't used in the style, enable attributes that are
+        var styleableShaderAttributes = content._styleableShaderAttributes;
+        if (defined(styleableShaderAttributes)) {
+            for (name in styleableShaderAttributes) {
+                if (styleableShaderAttributes.hasOwnProperty(name)) {
+                    attribute = styleableShaderAttributes[name];
+                    var enabled = (styleableProperties.indexOf(name) >= 0);
+                    var vertexAttribute = getVertexAttribute(vertexArray, attribute.location);
+                    vertexAttribute.enabled = enabled;
+                }
+            }
+        }
+
+        if (hasColorStyle) {
+            // Disable the color vertex attribute when a color style is used
+            var colorVertexAttribute = getVertexAttribute(vertexArray, colorLocation);
+            colorVertexAttribute.enabled = false;
+        }
+
+        var attributeLocations = {
+            a_position : positionLocation
+        };
+        if (hasColors && !hasColorStyle) {
+            attributeLocations.a_color = colorLocation;
+        }
+        if (hasNormals) {
+            attributeLocations.a_normal = normalLocation;
+        }
+        if (hasBatchIds) {
+            attributeLocations.a_batchId = batchIdLocation;
```

Comment:
I think I am missing something, but what are the batch id used for?  I thought each vertex attribute contained all the per-point properties so no batch id is needed to look up into a batch table since everything is in vertex attributes.